src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -122,6 +112,57 @@ public Socket createSocket() throws JedisConnectionException {
     }
   }
 
+  /**
+   * ssl enable check is done before this.
+   */
+  private Socket createSslSocket(HostAndPort _hostAndPort, Socket socket) throws IOException, GeneralSecurityException {
+
+    Socket plainSocket = socket;
+
+    SSLSocketFactory _sslSocketFactory;
+    SSLParameters _sslParameters;
+
+    if (sslOptions != null) {
+
+      SSLContext _sslContext = sslOptions.createSslContext();
+
+      _sslParameters = sslParameters != null ? sslParameters : new SSLParameters(); // TODO:
+
+      if (sslHostnameVerifyMode == SslHostnameVerifyMode.HTTPS) {
+        _sslParameters.setEndpointIdentificationAlgorithm("HTTPS");
+      } else if (sslHostnameVerifyMode == SslHostnameVerifyMode.DISABLE) {
+        _sslParameters.setEndpointIdentificationAlgorithm("");
+      }
+
+      _sslSocketFactory = _sslContext.getSocketFactory();
```

Comment:
nit: would put it close the context declaration, or even more make it single line